irods_specific_implementation/irods_rest_auth_api_implementation.h

```diff
@@ -5,31 +5,118 @@
 #define MACRO_IRODS_AUTH_API_IMPLEMENTATION \
     Pistache::Http::Code code; \
     std::string message; \
-    std::tie(code, message) = irods_auth_(user_name.get(), password.get(), auth_type.get()); \
+    std::tie(code, message) = irods_auth_(headers.getRaw("Authorization").value()); \
     response.send(code, message);
 
 namespace irods::rest {
 
     // this is contractually tied directly to the api implementation
     const std::string service_name{"irods_rest_cpp_auth_server"};
 
-    class auth : public api_base {
-
+    class auth : public api_base
+    {
         using json = nlohmann::json;
 
+        void throw_for_invalid_header(const std::string _h) {
+            THROW(SYS_INVALID_INPUT_PARAM,
+                  fmt::format("invalid Authorization Header {}", _h));
+        }
+
+		auto base64_decode(const std::string& _in)
+		{
+
+			static constexpr unsigned char table[] = {
+				64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
+				64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
+				64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 62, 64, 64, 64, 63,
+				52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 64, 64, 64, 64, 64, 64,
+				64,  0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14,
+				15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 64, 64, 64, 64, 64,
+				64, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40,
+				41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 64, 64, 64, 64, 64,
+				64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
+				64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
+				64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
+				64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
+				64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
+				64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
+				64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
+				64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64
+			};
+
+			auto in_len = _in.size();
+			if (in_len % 4 != 0) {
+                THROW(SYS_INVALID_INPUT_PARAM,
+                      "input is not a multiple of 4");
+            }
+
+			auto out_len = in_len / 4 * 3;
```

Comment:
indentation

Reply:
looks correct to me in the file, perhaps a tab/space issue in github?

Reply:
ran :retab on the file, should be fixed.